pkg/services/publicdashboards/service/service.go

```diff
@@ -190,38 +190,52 @@ func (pd *PublicDashboardServiceImpl) updatePublicDashboardConfig(ctx context.Co
 	return dto.PublicDashboard.Uid, pd.store.UpdatePublicDashboardConfig(ctx, cmd)
 }
 
-func (pd *PublicDashboardServiceImpl) GetQueryDataResponse(ctx context.Context, skipCache bool, reqDTO *PublicDashboardQueryDTO, panelId int64, accessToken string) (*backend.QueryDataResponse, error) {
-	if err := validation.ValidateQueryPublicDashboardRequest(reqDTO); err != nil {
-		return nil, ErrPublicDashboardBadRequest
+func (pd *PublicDashboardServiceImpl) GetQueryDataResponse(ctx context.Context, skipCache bool, queryDto PublicDashboardQueryDTO, panelId int64, accessToken string) (*backend.QueryDataResponse, error) {
+	metricReq, err := pd.GetMetricRequest(ctx, accessToken, panelId, queryDto)
+	if err != nil {
+		return nil, err
 	}
 
-	publicDashboard, dashboard, err := pd.GetPublicDashboard(ctx, accessToken)
+	_, dashboard, err := pd.GetPublicDashboard(ctx, accessToken)
```

Comment:
GetPublicDashboard is making an extra database call here. We're also calling this method in GetMetricRequest. I think we should consider either querying once at the top level of the service and passing in where needed or return have GetMetricRequest return the dashboard as well. I'm leaning toward the former.

Reply:
Revised: After looking at the enterprise-pr, I think it would be preferable to make an additional database method called GetDashboardByAccessToken which only makes one database call.